### DISCUSSION TITLE
select map on array value works with table object but not with the underscore operator

### DISCUSSION THREAD

**Discussion by ophiry** [ibis-project/ibis#9255] [Q&A] [answered]
for a table with a list field i'm trying to call map

    import ibis
    from ibis import _
    t = ibis.memtable({
        'i': [1,2,3], 
        'x': [[1,2],[3,5],[6,7] ]
      })
using the table object works:

    t.select(t.x.map(_ + 1))
but using the underscore api ([chain expressions][1]) fails:

    t.select(_.x.map(_ + 1))

> TypeError: unsupported operand type(s) for +: 'Table' and 'int'

cross posted from [stackoveflow](https://stackoverflow.com/questions/78533107/ibis-select-map-on-array-value-works-with-table-object-but-not-with-the-undersc):

  [1]: https://ibis-project.org/how-to/analytics/chain_expressions

**Comment by cpcloud** [comment, marked as answer]
Reposting my answer from StackOverflow:

The underscore API in this context would be ambiguous here as to whether it refers to a lambda parameter or the outer table.

As I say [here](https://github.com/ibis-project/ibis/issues/8553#issuecomment-1979554992), there's no real reason to prevent use of the second case in a call to `ArrayMap`.

For example:

```python
t.my_array_col.map(lambda x: _.a + x)
```

It's not clear how to reliably determine what the underscore means when mixed like in your original example, and we don't want to prevent valid uses of it in the underscore-as-outer-table use, so that's its meaning.
